components/atoms/Select/single-select.tsx

```diff
@@ -9,15 +10,50 @@ interface SingleSelectProps {
   value?: string;
   onValueChange: (value: string) => void;
   placeholder?: string;
+  inputPlaceholder?: string;
   options: { label: string; value: string }[];
   position?: "popper" | "item-aligned";
+  isSearchable?: boolean;
 }
 
-const SingleSelect = ({ placeholder, value, onValueChange, options, position }: SingleSelectProps) => {
+const SingleSelect = ({
+  placeholder,
+  value,
+  onValueChange,
+  options,
+  position,
+  inputPlaceholder,
+  isSearchable = false,
+}: SingleSelectProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [inputValue, setInputValue] = useState("");
+  const [isOpen, setIsOpen] = useState(false);
+
+  const filteredOptions = options.filter((option) => option.label.toLowerCase().includes(inputValue.toLowerCase()));
+
+  const getOptions = () => (isSearchable ? filteredOptions : options);
+
+  useEffect(() => {
+    if (isOpen && isSearchable) {
+      inputRef.current?.focus();
+    }
+  }, [isOpen, isSearchable]);
+
   return (
-    <Select value={value} onValueChange={(value) => onValueChange(value)}>
+    <Select
+      value={value}
+      onValueChange={(value) => {
+        onValueChange(value);
+      }}
```

Comment:
refactor (blocking): just set the prop to the function prop passed in.

```suggestion
      onValueChange={onValueChange}
```